packages/eas-cli/src/commandUtils/EasCommand.ts

```diff
@@ -127,14 +132,33 @@ export default abstract class EasCommand extends Command {
    */
   private readonly sessionManager = new SessionManager();
 
+  override get ctor(): typeof EasCommand {
+    return this.constructor as typeof EasCommand;
+  }
+
   protected abstract runAsync(): Promise<any>;
 
   // eslint-disable-next-line async-protect/async-suffix
   async run(): Promise<any> {
     await initAnalyticsAsync();
 
-    // this is needed for logEvent call below as it identifies the user in the analytics system
-    await this.sessionManager.getUserAsync();
+    // identify the user in the analytics system ahead of running the command
+    // to log run before any potential ctrl-c
+    let actor: Actor | undefined;
+    if ('loggedIn' in this.ctor.contextDefinition) {
+      const { flags } = await this.parse();
```

Comment:
What if this throws an error? 🤔 To clarify: I'm not sure when oclif validates input params.

Reply:
That's a really good point. I just checked and it does the validation here. It has been this way since it was written, but I'll try to find a better place to put this.